components/brave_wallet/common/zcash_utils.cc

```diff
@@ -225,19 +229,112 @@ std::optional<std::string> ExtractTransparentPart(
     return std::nullopt;
   }
 
-  auto parts = ParseUnifiedAddress(
+  auto parts = ParseUnifiedAddressBody(
       base::make_span(*reverted).subspan(0, reverted->size() - kPaddedHrpSize));
+
+  return parts;
+}
+
+// https://zips.z.cash/zip-0316#encoding-of-unified-addresses
+std::optional<std::string> ExtractTransparentPart(
+    const std::string& unified_address,
+    bool is_testnet) {
+  auto transparent_bytes = GetTransparentRawBytes(unified_address, is_testnet);
+  if (transparent_bytes) {
+    return PubkeyHashToTransparentAddress(transparent_bytes.value(),
+                                          is_testnet);
+  }
+  return std::nullopt;
+}
+
+std::optional<std::array<uint8_t, 43>> GetOrchardRawBytes(
+    const std::string& unified_address,
+    bool is_testnet) {
+  auto parts = ExtractParsedAddresses(unified_address, is_testnet);
+
+  if (!parts.has_value()) {
+    return std::nullopt;
+  }
+
+  for (const auto& part : parts.value()) {
+    if (part.first == ZCashAddrType::kOrchard) {
+      if (part.second.size() != 43) {
+        return std::nullopt;
+      }
+      std::array<uint8_t, 43> result;
```

Comment:
43 appears as magic number in many places now. Should be extracted to some constant